src/main/frontend/context/i18n.cljs

```diff
@@ -1,29 +1,27 @@
 (ns frontend.context.i18n
+  "Handles translation for the entire application. The dependencies for this ns
+  must be small since it is used throughout the application."
   (:require [frontend.dicts :as dicts]
-            [frontend.modules.shortcut.dict :as shortcut-dict]
-            [medley.core :refer [deep-merge]]
+            [frontend.modules.shortcut.dicts :as shortcut-dicts]
+            [tongue.core :as tongue]
             [frontend.state :as state]))
 
-;; TODO
-;; - [x] Get the preferred language from state
-;; - [x] Update the preferred language
-;; - [x] Create t functiona which takes a keyword and returns text with the current preferred language
-;; - [x] Add fetch for local browser preferred language if user has set it already
-;; - [ ] Fetch preferred language from backend if user is logged in
+(def dicts
+  (merge-with merge dicts/dicts shortcut-dicts/dicts))
 
-(defn fetch-local-language []
-  (.. js/window -navigator -language))
-
-(defonce translate-dicts (atom {}))
+(def translate
+  (tongue/build-translate dicts))
 
 (defn t
   [& args]
-  (let [preferred-language (keyword (state/sub :preferred-language))
-        _ (when (nil? preferred-language)
-            (state/set-preferred-language! (fetch-local-language)))
-        dicts (or (get @translate-dicts preferred-language)
-                  (let [result (some-> (deep-merge dicts/dicts shortcut-dict/dict)
-                                       dicts/translate)]
-                    (swap! translate-dicts assoc preferred-language result)
-                    result))]
-    (apply (partial dicts preferred-language) args)))
+  (let [preferred-language (keyword (state/sub :preferred-language))]
+    (apply translate preferred-language args)))
+
+(defn- fetch-local-language []
+  (.. js/window -navigator -language))
+
+;; TODO: Fetch preferred language from backend if user is logged in
+(defn start []
+  (let [preferred-language (state/sub :preferred-language)]
```

Comment:
We call this once at startup instead of N times where N is the number of translation calls for the lifetime of an app